cmd/api-endpoint/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/edulinq/autograder/internal/api"
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/exit"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+var args struct {
+	config.ConfigArgs
+	cmd.CommonOptions
+
+	Endpoint   string   `help:"Endpoint of the desired API." arg:""`
+	Parameters []string `help:"Parameter(s) for the endpoint" arg:"" optional:""`
+	Table      bool     `help:"Output data as a TSV." default:"false"`
+}
+
+func main() {
+	kong.Parse(&args,
+		kong.Description("Execute an API request to the specified endpoint."),
+	)
+
+	err := config.HandleConfigArgs(args.ConfigArgs)
+	if err != nil {
+		log.Fatal("Failed to load config options.", err)
+	}
+
+	var description core.EndpointDescription
+
+	describe := api.Describe(*api.GetRoutes())
+	for endpoint, requestResponse := range describe.Endpoints {
+		if endpoint == args.Endpoint {
+			description = requestResponse
+			break
+		}
+	}
+
+	if description == (core.EndpointDescription{}) {
+		log.Error("Failed to find the endpoint.", log.NewAttr("endpoint", args.Endpoint))
```

Comment:
You don't have to interact with the user through logs.
We log.Fatal() in CMDs as a quick way of exiting,
but if you are interacting with a user just printing is the standard way.
For example:
https://github.com/edulinq/autograder-server/blob/main/cmd/assignment-report/main.go
Notice we logged to exit quickly, but just printed normal information.